### PR TITLE
adding dist constants to Windows paths

### PR DESCRIPTION
**Pull request by bobchaos**
Signed-off-by: Marc Chamberland <chamberland.marc@gmail.com>

## Description
More work towards https://github.com/chef/chef/issues/8376 this one addresses config paths on  Windows systems. Turns out it's all done in one place, unless I'm misunderstanding something.

## Related Issue

https://github.com/chef/chef/issues/8376

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [X] I have read the **CONTRIBUTING** document.
- [X I have run the pre-merge tests locally and they pass.
- [NA] I have updated the documentation accordingly.
- [NA] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: chef-config/lib/chef-config/config.rb [hunk 1]
```diff
@@ -65,7 +65,7 @@ def self.platform_specific_path(path)
         # Some installations will be on different drives so use the drive that
         # the expanded path to __FILE__ is found.
         drive = windows_installation_drive
-        if drive && path[0] == '\\' && path.split('\\')[2] == "chef"
+        if drive && path[0] == '\\' && path.split('\\')[2] == Chef::Dist::Exec
```
Comment (tas50) [inline chef-config/lib/chef-config/config.rb]: while at the moment this path is the exec name we shouldn't count on that. We could change the executable name and this dir would stay the same ideall.

Comment (bobchaos) [reply chef-config/lib/chef-config/config.rb]: I'll make a Chef::Dist::WIN_CONF_DIR

Comment (bobchaos) [reply chef-config/lib/chef-config/config.rb]: I'll need to wait for https://github.com/chef/chef/pull/9060 to be merged, the new constant should go in `ChefConfig::Dist` but #9060 creates it.

**Comment by bobchaos** [comment]
This is now obsoleted by https://github.com/chef/chef/pull/9068, closing

**Pull request closed without merging**